src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -277,12 +279,66 @@ internal unsafe SocketError DoOperationSend(SafeCloseSocket handle)
 
         private void InnerStartOperationSendPackets()
         {
-            throw new PlatformNotSupportedException();
+            // nop
         }
 
         internal SocketError DoOperationSendPackets(Socket socket, SafeCloseSocket handle)
         {
-            throw new PlatformNotSupportedException();
+            Debug.Assert(_sendPacketsElements != null);
```

Comment:
To me asserts like this are about communicating clearly to someone reading the code certain expectations.  I see that assert and I think "ok, the developer who wrote this is telling me that this won't be null, therefore it's fine to assume it's not, and the developer is confident about that because they must be verifying it or guaranteeing it by construction elsewhere".  Without the assert, I don't know.  Obviously that's not necessary in all cases, but for something like this, where the guarantee is made in a different function, in a different part of the code, I personally find value in it.